### PR TITLE
Warnings if unable to config interface

### PR DESCRIPTION
**Pull request by TMRh20**
- Add some warnings with detailed descriptions of the problem and what to do if configuration of the network interface fails (usually because not running as root)

### CODE REVIEW

**Comment by 2bndy5** [comment]
I think we can programmatically fetch the user name using [`getlogin()`](https://pubs.opengroup.org/onlinepubs/9699919799/functions/getlogin.html).

**Comment by TMRh20** [comment]
Great idea! I've added `getlogin()` calls to modify the commands provided to the user now!

File: RF24Gateway.cpp [hunk 1]
```diff
@@ -209,7 +209,12 @@ int ESBGateway<mesh_t, network_t, radio_t>::allocateTunDevice(char* dev, int fla
         // close(fd);
         //#if (RF24GATEWAY_DEBUG_LEVEL >= 1)
         std::cerr << "RF24Gw: Error: enabling TUNSETIFF" << std::endl;
-        std::cerr << "RF24Gw: If changing from TAP/TUN, run 'sudo ip link delete tun_nrf24' to remove the interface" << std::endl;
+        uint32_t UID = getuid();
+        if (UID) {
```
Comment (2bndy5) [inline RF24Gateway.cpp:213]: I've been reading up on [`getuid()`](https://pubs.opengroup.org/onlinepubs/9699919799/functions/getuid.html). Am I to understand that a zero returned indicates super user or root?

Comment (TMRh20) [reply RF24Gateway.cpp:213]: Yup, root is 0

**Review by 2bndy5 — APPROVED** [review]

**Pull request merged**